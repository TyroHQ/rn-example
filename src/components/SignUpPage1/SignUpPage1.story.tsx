import { storiesOf } from "@storybook/react-native";
import figmaDecorator from "storybook-addon-figma";

import * as usage from "./SignUpPage1.usage";

let figma = undefined;

let story = storiesOf("SignUpPage1", module);

if (figma) {
  story = story.addDecorator(
    figmaDecorator({
      url: figma
    })
  );
}

for (const example in usage) {
  story = story.add(example, usage[example]);
}
