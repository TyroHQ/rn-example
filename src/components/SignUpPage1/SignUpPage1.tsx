import React from "react";
import { Text } from "react-native";
import styled from "styled-components/native";
import { Placeholder } from "../Placeholder";

const SignUpPage1Container = styled.View`
  background-color: lightgrey;
  padding: 10px;
`;

export class SignUpPage1 extends React.Component<{}> {
  static defaultProps: SignUpPage1["props"];
  render() {
    return (
      <Placeholder>
        Sign up page container
        <Placeholder>Header bar</Placeholder>
        <Placeholder>
          Content Area
          <Placeholder>Sign Up Now Header</Placeholder>
          <Placeholder>First Name Entry Field</Placeholder>
          <Placeholder>Last Name Entry Field</Placeholder>
          <Placeholder>Password Entry Field</Placeholder>
          <Placeholder>Confirm Password Entry Field</Placeholder>
          <Placeholder>Next Button</Placeholder>
        </Placeholder>
      </Placeholder>
    );
  }
}
