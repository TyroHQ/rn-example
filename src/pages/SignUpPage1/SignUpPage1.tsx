import React from "react";
import { Text, View } from "react-native";
import styled from "styled-components";
import { Placeholder } from "@src/components/Placeholder";

const SignUpPage1Container = styled(View)`
  background-color: lightgrey;
  padding: 10px;
`;

export class SignUpPage1 extends React.Component<{}> {
  static defaultProps: SignUpPage1["props"];
  render() {
    return (
      <Placeholder>
        Sign up page container
        <Placeholder>Header bar</Placeholder>
        <Placeholder>
          Content Area
          <Placeholder>Sign Up Now Header</Placeholder>
          <Placeholder>First Name Entry Field</Placeholder>
          <Placeholder>Last Name Entry Field</Placeholder>
          <Placeholder>Password Entry Field</Placeholder>
          <Placeholder>Confirm Password Entry Field</Placeholder>
          <Placeholder>Next Button</Placeholder>
        </Placeholder>
      </Placeholder>
    );
  }
}
