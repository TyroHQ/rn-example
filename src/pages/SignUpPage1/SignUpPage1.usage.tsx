import React from "react";
import { boolean, text, number } from "@storybook/addon-knobs";
import { action } from "@storybook/addon-actions";
import { SignUpPage1 } from "./";

export const basic = () => <SignUpPage1 />;
