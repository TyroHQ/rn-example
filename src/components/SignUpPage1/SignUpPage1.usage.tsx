import React from "react";
import { boolean, text, number } from "@storybook/addon-knobs";
import { action } from "@storybook/addon-actions";
import { SignUpPage1 } from "./SignUpPage1";

export const basic = () => <SignUpPage1 />;
