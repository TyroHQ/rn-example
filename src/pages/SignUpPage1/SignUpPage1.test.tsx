import { shallow } from "enzyme";
import * as usage from "./SignUpPage1.usage";

describe("<SignUpPage1 />", () => {
  for (const example in usage) {
    it(`renders well in ${example} mode`, () => {
      const x = (usage as any)[example]();
      const wrapper = shallow(x);
      expect(wrapper).toMatchSnapshot();
    });
  }
});
