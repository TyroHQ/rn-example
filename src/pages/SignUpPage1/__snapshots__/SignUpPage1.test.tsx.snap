// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUpPage1 /> renders well in basic mode 1`] = `
<Placeholder
  orientation="row"
>
  Sign up page container
  <Placeholder
    orientation="row"
  >
    Header bar
  </Placeholder>
  <Placeholder
    orientation="row"
  >
    Content Area
    <Placeholder
      orientation="row"
    >
      Sign Up Now Header
    </Placeholder>
    <Placeholder
      orientation="row"
    >
      First Name Entry Field
    </Placeholder>
    <Placeholder
      orientation="row"
    >
      Last Name Entry Field
    </Placeholder>
    <Placeholder
      orientation="row"
    >
      Password Entry Field
    </Placeholder>
    <Placeholder
      orientation="row"
    >
      Confirm Password Entry Field
    </Placeholder>
    <Placeholder
      orientation="row"
    >
      Next Button
    </Placeholder>
  </Placeholder>
</Placeholder>
`;
